lucet-runtime/lucet-runtime-internals/src/region/mmap.rs

```diff
@@ -376,16 +380,35 @@ impl MmapRegion {
 
         // lay out the other sections in memory
         let heap = mem as usize + instance_heap_offset();
-        let stack = heap + region.limits.heap_address_space_size + host_page_size();
+        let stack_guard = heap + region.limits.heap_address_space_size;
+        let stack = stack_guard + host_page_size();
         let globals = stack + region.limits.stack_size;
         let sigstack = globals + region.limits.globals_size + host_page_size();
+        let stack_emergency_guard = sigstack + region.limits.signal_stack_size;
+        let stack_emergency_space = stack_emergency_guard + host_page_size();
+
+        // ensure we've accounted for all space
+        assert_eq!(
+            stack_emergency_space + Slot::emergency_stack_size() - mem as usize,
+            region.limits.total_memory_size()
+        );
+
+        // set the emergency stack to read/write, so it's ready in advance in case we need it.
```

Comment:
Is there a scenario where this would be needed before the sigaltstack is unprotected? If not I'd rather treat them uniformly.